src/PSR7/Validators/BodyValidator/UnipartValidation.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenAPIValidation\PSR7\Validators\BodyValidator;
+
+use cebe\openapi\spec\MediaType;
+use OpenAPIValidation\PSR7\Exception\Validation\InvalidBody;
+use OpenAPIValidation\PSR7\OperationAddress;
+use OpenAPIValidation\Schema\Exception\SchemaMismatch;
+use OpenAPIValidation\Schema\SchemaValidator;
+use Psr\Http\Message\MessageInterface;
+use const JSON_ERROR_NONE;
+use function json_decode;
+use function json_last_error;
+use function json_last_error_msg;
+use function preg_match;
+
+trait UnipartValidation
+{
+    /**
+     * @param MediaType[] $mediaTypeSpecs
+     *
+     * @throws InvalidBody
+     */
+    private function validateUnipart(OperationAddress $addr, MessageInterface $message, array $mediaTypeSpecs, string $contentType) : void
+    {
+        if (preg_match('#^application/.*json$#', $contentType)) {
```

Comment:
It looks like this part is replicated everywhere now and really not extensible. if you would add, i.e xml support then you would patch multiple place.

Reply:
I think we need some kind of body extractor here. it's actually a kind of deserializer in the end per content-type mask. i.e we can have interface for 
```
public function extractBody(/** stream */ $body, string $contentType): object;

public function supports(string $contentType): bool
```

which can be extended by iterating with all registered extractors, like

```php
final class DelegatingBodyExtractor implements BodyExtractor 
{
  private $extractors;

  public function extractBody(/** stream */ $body, string $contentType): object
  { 
     foreach ($this->extractors as $extractor) {
        if (! $extractor->supports($contentType) {
           continue;
        }

        return $extractor->extractBody($body, $contentType);
     }
 
     throw new \InvalidArgumentException('Content-Type is not supported')
  }

  public function supports(string $contentType): bool
  {
     foreach ($this->extractors as $extractor) {
        if ($extractor->supports($contentType) {
           return true;
        }
     }

     return false;
  }
}

Reply:
Excellent idea!